### ISSUE TITLE
Specify versions in requirements.txt

### ISSUE THREAD

**Issue by 2ec** [tloen/alpaca-lora#200]
I have had a couple of errors trying to get this repo running, and it seems that using bitsandbytes==0.37.0 solved it for me, instead of the 0.37.2 that pip downloads. 

I also had to use the trick from @moorehousew in https://github.com/tloen/alpaca-lora/issues/46#issuecomment-1474343963
`cp libbitsandbytes_cuda117.so libbitsandbytes_cpu.so` to get it working.

Guess it will be useful for others to have the correct versions specified to minimize debug questions🤷‍♂️

**Comment by claysauruswrecks** [comment]
@2ec - Can you post your version-locked `requirements.txt`?

I think we should keep the regular `requirements.txt` containing only the top-level requirements with no version numbers, and a `requirements-lock.txt` which has all packages and their latest working versions.

**Comment by FrancescoSaverioZuppichini** [comment]
Also `torch==2.1.0` works 

You can also use my dockerfile 

https://github.com/tloen/alpaca-lora/pull/91

**Comment by mirekphd** [comment]
TL;DR I'd skip pinning packages with equality signs. If you must than allow for security updates by specifying a range using two inequalities: a greater or equal than the currently working version and less than next major version (to guard against breaking changes assuming semantic versioning is followed).

It is every security department's nightmare and can be justified only when moving to production particularly brittle and hard to reproduce data science output (e.g. if your models are not in a future-proof format but instead say, pickled which pins them down to the current python version and maybe also trained using some high-level wrapper that dumps from memory binary versions of its frequently-changed unstable classes to the model file). 

From my 5-year experience building data science and ML containers this can only lead to untold misery and is used as a temporary defense against breaking changes in some deeply nested API (a must if your builds are more frequent than downstream community reaction times to such changes:)
